### PR TITLE
CV32A6 CSR Access dvplan

### PR DESCRIPTION
**Pull request by spidugu444**
Hi All,

This PR contains dvplan for  CV32A6 CSR Access Verification and markdown version of dvplan has also been added.
 
Let me know if any changes are required.

Thanks,

### CODE REVIEW

File: cva6/docs/VerifPlans/csr_access/VP_IP003.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          \nThe functionality of user mode counter CSR is being tested by performing
+          two continuous reads and checking whether the value in the second read is
+          greater than the value in the first read."
+        reqt_doc: 
+          https://docs.openhwgroup.org/projects/cva6-user-manual/01_cva6_user/CV32A6_Control_Status_Registers.html
+        ref_mode: page
+        ref_page: ''
+        ref_section: ''
+        ref_viewer: firefox
+        verif_goals: "1.Verify that these CSR are properly initialized.\n2.Initiate
```
Comment (MikeOpenHWGroup) [inline cva6/docs/VerifPlans/csr_access/VP_IP003.yml:56]: - It is not possible to check that the `cycle` and `instret` CSRs are properly initialized using a test-program running on the core.
- Please make a note that this algorithm is only an "approximate test" of the functionality of these CSRs.

File: cva6/docs/VerifPlans/csr_access/VP_IP003.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          a second read from the counter CSR immediately after the first read.\n3.Ensure
+          that the value of the second read from counter CSR is greater than the value
+          of the initial read.\n4.Confirm that counter CSRs are correctly incrementing."
```
Comment (MikeOpenHWGroup) [inline cva6/docs/VerifPlans/csr_access/VP_IP003.yml:59]: We cannot say that we can "Confirm that counter CSRs are _correctly_ incrementing", only that the CSRs _are_ incrementing.

File: cva6/docs/VerifPlans/csr_access/VP_IP004.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
+!Feature
+next_elt_id: 4
+name: CVA6_Machine_mode_counter_csr(mcycle,mcycleh,minstret,minstreth)
```
Comment (MikeOpenHWGroup) [inline cva6/docs/VerifPlans/csr_access/VP_IP004.yml:3]: The same comments are per the `CVA6_User_Mode_Counter_CSRs(cycle, instret, cycleh, instreth)` CSRs will apply here.

**Review by MikeOpenHWGroup — COMMENTED** [review]
Hi @spidugu444 this is a good start to the CSR DVplan.  Please see my embedded comments.

File: cva6/docs/VerifPlans/csr_access/VP_IP001.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        reqt_doc: 
+          https://docs.openhwgroup.org/projects/cva6-user-manual/01_cva6_user/CV32A6_Control_Status_Registers.html
+        ref_mode: page
+        ref_page: ''
+        ref_section: ''
+        ref_viewer: firefox
+        verif_goals: "1.Ensure that Machine mode read only CSRs can only be accessed
+          in Machine mode according to the RISC-V specification.\n\n2.Verify that
+          trying to access a Machine read only CSRs in an lower privilege mode raises
+          an illegal instruction exception."
```
[the remark this answers is not in the document]
Comment (spidugu444) [reply cva6/docs/VerifPlans/csr_access/VP_IP001.yml]: Added above mentioned line.

File: cva6/docs/VerifPlans/csr_access/VP_IP002.yml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        name: '000'
+        tag: VP_csr-access_F004_S003_I000
+        description: Check the behaviour of the RISC-V Supervisor mode CVA6 CSRs,when
+          reset inverted values are written to respective CSRs.
+        reqt_doc: ''
+        ref_mode: page
+        ref_page: ''
+        ref_section: ''
+        ref_viewer: firefox
+        verif_goals: "1. Verify CSR reading post write operation.\n2. Verify if the
```
[the remark this answers is not in the document]
Comment (spidugu444) [reply cva6/docs/VerifPlans/csr_access/VP_IP002.yml]: Removed inverted POR value sub-feature and added inverted POR value feature in the list of random values.

File: cva6/docs/VerifPlans/csr_access/VP_IP002.yml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+- 002_CSR write and read operations: !Subfeature
+    name: 002_CSR write and read operations
+    tag: VP_CSR_VERIFICATION_F004_S001
+    next_elt_id: 1
+    display_order: 2
+    items: !!omap
+    - '000': !VerifItem
+        name: '000'
+        tag: VP_CSR_VERIFICATION_F004_S001_I000
+        description: Check the correctness of RISCV CVA6 Supervisor Mode RW CSR by
```
[the remark this answers is not in the document]
Comment (spidugu444) [reply cva6/docs/VerifPlans/csr_access/VP_IP002.yml:42]: Removed inverted POR value sub-feature and added inverted POR value feature in the list of random values.

File: cva6/docs/VerifPlans/csr_access/VP_IP003.yml (hunk 1 above)
Comment (spidugu444) [reply cva6/docs/VerifPlans/csr_access/VP_IP003.yml:56]: Added above mentioned Note at the end of verif_goals.

File: cva6/docs/VerifPlans/csr_access/VP_IP003.yml (hunk 2 above)
Comment (spidugu444) [reply cva6/docs/VerifPlans/csr_access/VP_IP003.yml]: Modified verif_goals as mentioned above i.e **Confirm that counter CSRs are incrementing.**

File: cva6/docs/VerifPlans/csr_access/VP_IP004.yml (hunk 3 above)
Comment (spidugu444) [reply cva6/docs/VerifPlans/csr_access/VP_IP004.yml:3]: Modified machine Machine mode counters.

**Comment by MikeOpenHWGroup** [comment]
Hi @spidugu444, thanks for fixing the CI.  Please have a look at my outstanding comments on `IP002`.

**Pull request closed without merging**